DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -212,13 +275,106 @@ namespace edmNew {
       typename DetSetVector<T>::Item & item;
       bool saveEmpty;
     };
+
+    /* fill on demand a given  DetSet
+     */
+    class TSFastFiller {
+    public:
+      typedef typename DetSetVector<T>::data_type value_type;
+      typedef typename DetSetVector<T>::id_type key_type;
+      typedef typename DetSetVector<T>::id_type id_type;
+      typedef typename DetSetVector<T>::size_type size_type;
+
+#ifdef USE_ATOMIC
+      static DetSetVector<T>::Item & dummy() {
+        static  DetSetVector<T>::Item d; return d;
+      }
+      TSFastFiller(DetSetVector<T> & iv, id_type id) :
+        v(iv), item(v.ready()? v.push_back(id): dummy()) { assert(v.filling==true); v.filling = false;}
```

Comment:
Could you add a comment explaining why ```v.filling``` is required to be ```true``` here? I'm having a hard time understanding the synchronization logic.